javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java

```diff
@@ -227,8 +226,11 @@ private static void installEarlyInstrumentation(
     AgentBuilder.Identified.Extendable extendableAgentBuilder =
         agentBuilder
             .ignore(
-                target -> instrumentationInstalled, // turn off after instrumentation is installed
-                Objects::nonNull)
```

Comment:
I removed the `Objects::nonNull`(checks that instrumentation applies only in boot loader) because it doesn't work as intended. When both type and class loader matcher are given both must match for the type to be ignored, as far as I understand here the intent was to ignore when either matches. If we want we can add the class loader check with `.or(Objects::nonNull)` but it shouldn't be necessary.